crates/primitives/src/checkpoints.rs

```diff
@@ -194,6 +229,8 @@ pub enum StageUnitCheckpoint {
     Account(AccountHashingCheckpoint),
     /// Saves the progress of StorageHashing stage.
     Storage(StorageHashingCheckpoint),
+    /// Saves the progress of Hashing stage.
+    Headers(HeadersCheckpoint),
```

Comment:
I like different checkpoint types that are specific to the changes.
This deviates from the generalized "stage" abstraction, but is easier to understand and we only have a limited number of stages